plugins/parsers/grok/parser.go

```diff
@@ -137,12 +142,32 @@ func (p *Parser) Compile() error {
 	p.NamedPatterns = make([]string, 0, len(p.Patterns))
 	for i, pattern := range p.Patterns {
 		pattern = strings.TrimSpace(pattern)
-		if pattern == "" {
-			continue
+
+		//checks that there is only one named field in pattern and 2 ':' indicating a modifier
+		//then extract any modifiers off pattern
+		if strings.Count(pattern, "%") == 1 && strings.Count(pattern, ":") == 2 {
+			pattern = strings.Trim(pattern, "%{}")
```

Comment:
This will leave a `}` (`strings.Trim("%{doggy}:dog:perro", "%{}")` returns `doggy}:dog:perro`), move this after `SplitN` and change it to trim `splitPattern[0]`

Reply:
Actually, i would just remove this `.Trim` and not re-add the `%{` and `}` characters on line 158.

Reply:
so i was under the impression that this form: `%{doggy}:dog:perro` is not supported by the grok parser at all.  It is necessary to trim the `%{}` characters because if not, the tag will be added as `tag}`.